### ISSUE TITLE
Explore Zeppelin notebook to develop visualizations for hackathon

### ISSUE THREAD

**Issue by hyi** [ResearchSoftwareInstitute/greendatatranslator#26]
- 100 % complete

**Comment by stevencox** [comment]
Hi Hong, I've been installing and configuring [Lightning](https://stars-lightning.renci.org/).

It's got reasonably good [documentation](http://lightning-viz.org/visualizations/adjacency/). 

This note has an example [usage](https://stars-zeppelin.renci.org/#/notebook/2CA23SKUS) at the bottom. Let  me know if you'd like to discuss.

On a separate note, @cpschmitt and I discussed a potentially useful idea regarding visualizing Semantic Web data sets. Again, let me know if you'd like to discuss.

**Comment by hyi** [comment]
@stevencox Yes, Steve. I'm definitely interested in discussing these with you. Let's touch base on this offline. I will email you.

**Comment by hyi** [comment]
@stevencox and I talked about the idea he and @cpschmitt discussed for visualizing semantic web data set and we agreed I can explore this idea in Zeppelin to get started with visualization. I have created a SemanticDataVisualization note in Zeppelin here:  https://stars-zeppelin.renci.org/#/notebook/2CBKC7VQZ from which you can input n3 RDF data (default is to use drugbank.n3 in chem2bio2rdf datasets) and search term (default is 'Compound'), and a node link list will be extracted that is suitable to do a graph visualization. I also explored a bit on Lightning http://lightning-viz.org/ @stevencox has installed for visualization, and found although it has good performance, the API for force-directed graph visualization is limited in terms of customization to fit user's specific needs, so I am planning to explore bokeh http://bokeh.pydata.org/en/latest/ next to see if it works well for us.

**Comment by hyi** [comment]
Have explored Bokeh and found it does not have native graph rendering built in yet, so it will not serve our visualization need for now. Decided to go with D3 JavaScript library for graph visualization in Zeppelin notebook environment. The plan is to look at semantic data triples coming from different data sources, and look at the namespaces in URIs and match triples form different data sources to create a data source connection graph, and create visualization of how different data sources are connected.

**Comment by hyi** [comment]
Have created interactive visualization in Zeppelin notebook environment using D3 JavaScript library that shows linkages between subjects and objects that match a certain search term (e.g., compound) in a RDF data set (e.g., drugbank.n3). It shows the subjects/objects when hovering over a node and shows predicates when hovering over an edge/link. See a screen capture below. I initially planned to write a script to be run on demand on stars directly to traverse into data sources we are interested in, match triple terms based on our criteria to create summary relationships across different data sources, then output the results into a triple store n3 file. Then this output n3 file can be visualized using the visualization I created in Zeppelin. However, @stevencox found an existing visualization tool called LD-VOWL and have the tool pointed to the SPARQL query on our Balzegraph instance as shown here: http://vowl.visualdataweb.org/ldvowl/#/graph?endpointURL=http:%2F%2Fstars-blazegraph.renci.org%2Fbigdata%2Fsparql that dynamically discovers and visualizes our Blazegraph instance. However, it is not picking up more data and is limited for interactive exploration of the linked data. After discussing with @stevencox, we decided for me to look into LD-VOWL source code to see if I can customize the tool to fit our needs so that we can better understand semantic relationships between different data sources in our Blazegraph instance and use it as a data model.
![zeppelin_viz](https://cloud.githubusercontent.com/assets/476302/24614412/2022f81a-1859-11e7-8498-d62c1967cb0d.png)
